client/lib/tree-select/index.js

```diff
@@ -47,8 +54,19 @@ export default function treeSelect( getDependents, selector ) {
 		leafCache.set( key, value );
 		return value;
 	};
+
+	cachedSelector.clearCache = () => {
+		cache = new WeakMap();
+	};
+
+	return cachedSelector;
 }
 
+/*
+ * This object will be used as a WeakMap key if a dependency is a falsy value (null, undefined, ...)
+ */
+const STATIC_FALSY_KEY = {};
```

Comment:
I actually liked the `MixedMap` @samouri originally had in here. It had clear semantics and didn't require these odd workarounds. Is there real opposition against using `MixedMap`?

Reply:
The semantics now is that the dependents are of type `Maybe<Object>` -- that looks quite clean to me. If I encounter a place where I need the `MixedMap`, I'll start rooting for it.

Reply:
If we only have a need to support null/undefined/false, then i don't think we should incorporate MixedMap which allowed for unbounded primitives like int